common/txmgr/types/models.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+type TxmErrorType int
+
+// Generalized transaction manager error types that dictates what should be the next action, depending on the RPC error response.
+const (
+	Successful TxmErrorType = iota
+	Fatal
+	Retryable
+	Underpriced
+	Unknown
+	Unsupported
+	SuccessfulMissingReceipt
```

Comment:
The name is slightly confusing.
Rename this error type to TransactionAlreadyKnown.
Also add a comment explaining this error can be returned even if another Tx using the same sequenceId was already mined.